### PR TITLE
feat(nimtable): reuse existing env and add risedev nimtable

### PR DESCRIPTION
**Pull request by chenzl25**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- reuse AWS_REGION, AWS_S3_BUCKET, RW_DATA_DIRECTORY, RW_SQL_ENDPOINT, RW_SQL_DATABASE, RW_SQL_USERNAME and RW_SQL_PASSWORD, RW_BACKEND env.
- Add `./risedev d nimtable` to simplify the startup process

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/connector/src/sink/iceberg/mod.rs [hunk 1]
```diff
@@ -152,6 +152,9 @@ pub struct IcebergConfig {
 
     #[serde(default, deserialize_with = "deserialize_bool_from_string")]
     pub create_table_if_not_exists: bool,
+
+    #[serde(default, deserialize_with = "deserialize_bool_from_string")]
```
Comment (Li0k) [inline src/connector/src/sink/iceberg/mod.rs:156]: just a question: Is there a situation where we create non-nimtable services in a branch?

This config item is a Bool and I think it should have corresponding prefixes like ‘is_xxx’, ‘enable_xxx’.... .etc

File: src/connector/src/sink/iceberg/mod.rs [hunk 2]
```diff
@@ -320,7 +375,82 @@ impl IcebergConfig {
...
+                            "Invalid s3 path: {}, bucket is missing",
+                            self.path
+                        ))
+                    })?
+                    .to_string();
+                let root = url.path().trim_start_matches('/').to_string();
+                (bucket, root)
+            };
+
+            iceberg_configs.insert("iceberg.table.io.bucket".to_string(), bucket);
```
Comment (Li0k) [inline src/connector/src/sink/iceberg/mod.rs:417]: Observe L395, specialise the region path of `icelake` and `iceberg-rust`. May I ask if the bucket path needs the same treatment? Can you provide the link in the doc?

File: src/connector/src/source/iceberg/mod.rs [hunk 3]
```diff
@@ -69,6 +69,13 @@ pub struct IcebergProperties {
     #[serde(rename = "catalog.jdbc.password")]
     pub jdbc_password: Option<String>,
 
+    #[serde(
+        rename = "nimtable",
+        default,
+        deserialize_with = "deserialize_optional_bool_from_string"
+    )]
+    pub nimtable: Option<bool>,
```
Comment (Li0k) [inline src/connector/src/source/iceberg/mod.rs:77]: ditto

**Review by Li0k — APPROVED** [review]
Rest LGTM, thanks for the efforts

File: src/connector/src/sink/iceberg/mod.rs (hunk 2 above)
Comment (chenzl25) [reply src/connector/src/sink/iceberg/mod.rs:417]: The reason why bucket is needed is both icelake and iceberg-rust use opendal under the hood, so bucket is needed. However, the region is handled differently by icelake and iceberg-rust. They use different name to parse the config map.

File: src/connector/src/sink/iceberg/mod.rs (hunk 1 above)
Comment (chenzl25) [reply src/connector/src/sink/iceberg/mod.rs:156]: Good point. I think using `enable_config_load` is a better name.

**Pull request merged**